bids/modeling/statsmodels.py

```diff
@@ -686,6 +691,23 @@ def __init__(self, node, entities={}, collections=None, inputs=None,
         self.data = df.loc[:, var_names]
         self.metadata = df.loc[:, df.columns.difference(var_names)]
 
+        # replace missing values with 0 and warn user which columns had missing values
+        if missing_values == 'fill':
+            missing_cols = self.data.columns[self.data.isnull().any()]
+            if len(missing_cols) > 0:
+                warnings.warn("The following columns had missing values and "
+                              f"were filled with 0: {missing_cols}."
+                                "Consider explicitly handling missing values "
+                                "using transformations.")
+        
+            self.data = self.data.fillna(0)
+        elif missing_values == 'error':
+            if self.data.isnull().values.any():
+                raise ValueError("Missing values found in data matrix. "
```

Comment:
I would go ahead and list the missing columns in both cases. Might as well be helpful. Possibly:

```Python
if missing_values != 'ignore':
    missing_cols = self.data.columns[self.data.isnull().any()]
    if len(missing_cols) > 0:
        base_message = f"NaNs detected in columns: {missing_cols}"
        if missing_values == 'fill':
            ...
        elif missing_values == 'error':
            ...
```